projects/legacy/components/input-phone/utils/complete-phone-insertion-preprocessor.ts

```diff
@@ -40,12 +40,13 @@ export function tuiCreateCompletePhoneInsertionPreprocessor(
                      * and mask are ready to reject "extra" characters.
                      * We should cut leading country prefix to save trailing characters!
                      */
-                    countDigits(value) > completePhoneLength
+                    countDigits(value) > completePhoneLength ||
+                    value.startsWith(countryCode)
```

Comment:
See failed screenshot test

![image](https://github.com/user-attachments/assets/0d5a2f63-94de-41d7-955c-3da6bdb05b97)

Reply:
Probably, we should add new condition `|| value.startsWith(countryCode)` only for `data` property

Reply:
I agree! as that block only takes care of number processing on paste events.

Resolved!